Formula/k/kubehound.rb

```diff
@@ -0,0 +1,37 @@
+class Kubehound < Formula
+  desc "Tool for building Kubernetes attack paths"
+  homepage "https://kubehound.io"
+  url "https://github.com/DataDog/KubeHound/archive/refs/tags/v1.3.3.tar.gz"
+  sha256 "5ee3ca8621d4611e682ed2325bb12b87d6f00c4f7717195d5f43856f5d9b7917"
+  license "Apache-2.0"
+
+  depends_on "go" => [:build, :test]
+  depends_on "docker" => :test
+
+  def install
+    goos = Utils.safe_popen_read("#{Formula["go"].bin}/go", "env", "GOOS").chomp
+    goarch = Utils.safe_popen_read("#{Formula["go"].bin}/go", "env", "GOARCH").chomp
+
+    ldflags = %W[
+      -s -w
+      -X github.com/DataDog/KubeHound/pkg/config.BuildVersion=v#{version}
+      -X github.com/DataDog/KubeHound/pkg/config.BuildBranch=main
+      -X github.com/DataDog/KubeHound/pkg/config.BuildOs=#{goos}
+      -X github.com/DataDog/KubeHound/pkg/config.BuildArch=#{goarch}
+    ]
+    system "go", "build", *std_go_args(ldflags:), "-o", "./bin/build/kubehound", "./cmd/kubehound/"
+    bin.install "bin/build/kubehound" => "kubehound"
+  end
+
+  test do
+    # Check that the binary is working
+    goos = Utils.safe_popen_read("#{Formula["go"].bin}/go", "env", "GOOS").chomp
+    goarch = Utils.safe_popen_read("#{Formula["go"].bin}/go", "env", "GOARCH").chomp
+    assert_equal "kubehound version: v#{version} (#{goarch}/#{goos})", shell_output("#{bin}/kubehound version")
+
+    # Docker is required at runtime for some command but test environment does not permit such tests
+    ENV["DOCKER_HOST"] = "unix://#{testpath}/invalid.sock"
```

Comment:
does that mean that there is no docker dependency for tests?

Reply:
I initially didn't set the `depends_on "docker"` line and the test were running locally, probably because it was using the `docker` from my laptop.
Unfortunately it failed the CI, and I assumed it was missing the dependency.

I searched through other formulas the one with `depends_on "docker"`, and [all of them](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-core+%22depends_on+%5C%22docker%5C%22%22&type=code) actually set the docker host socket to invalid, so I assumed the docker socket was made unavailable from inside the test, probably for security reasons, and that tests depending on docker could simply not be launched.
I didn't find online documentation to explain how to run such tests. If you have any information, I would be happy to look at it and change this

I removed the docker dependency, please tell me if I should add it back :)